### PR TITLE
Fix detector peak predictor to include Peak Intensity Threshold

### PR DESCRIPTION
**Pull request by darshdinger**
As discussed with Robert and Malcolm, this commit includes changes to:
DetectorPeakPredictor to incorporate a new parameter for Peak intensity threshold
DiffractionSpectrumWeightCalculator to set the new peak intensity threshold parameter to 0.00

### CODE REVIEW

File: src/snapred/backend/recipe/algorithm/DiffractionSpectrumWeightCalculator.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def PyExec(self):
                 "Predicting peaks...",
                 InstrumentState=self.getProperty("InstrumentState").value,
                 CrystalInfo=self.getProperty("CrystalInfo").value,
+                PeakIntensityThreshold=0.00,
```
Comment (bolotovskyr) [inline src/snapred/backend/recipe/algorithm/DiffractionSpectrumWeightCalculator.py:39]: You want to set this as a string: 
`PeakIntensityThreshold="0"`

Comment (darshdinger) [reply src/snapred/backend/recipe/algorithm/DiffractionSpectrumWeightCalculator.py]: Done!

File: src/snapred/backend/recipe/algorithm/DetectorPeakPredictor.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ def PyInit(self) -> None:
         # declare properties
         self.declareProperty("InstrumentState", defaultValue="", direction=Direction.Input)
         self.declareProperty("CrystalInfo", defaultValue="", direction=Direction.Input)
+        self.declareProperty("PeakIntensityThreshold", defaultValue="0.05", direction=Direction.Input)
```
Comment (peterfpeterson) [inline src/snapred/backend/recipe/algorithm/DetectorPeakPredictor.py:21]:
```suggestion
        self.declareProperty("PeakIntensityThreshold", defaultValue=0.05, direction=Direction.Input)
```
will make the property a float so you don't have to convert it

File: src/snapred/backend/recipe/algorithm/DetectorPeakPredictor.py [hunk 3]
```diff
@@ -36,7 +37,7 @@ def PyExec(self) -> None:
         multiplicity = np.array(crystalInfo.multiplicities)
         dSpacing = np.array(crystalInfo.dSpacing)
         A = fSquared * multiplicity * dSpacing**4
-        thresholdA = np.max(A) * 0.05
+        thresholdA = np.max(A) * float(self.getProperty("PeakIntensityThreshold").value)
```
Comment (peterfpeterson) [inline src/snapred/backend/recipe/algorithm/DetectorPeakPredictor.py:40]:
```suggestion
        thresholdA = np.max(A) * self.getProperty("PeakIntensityThreshold").value
```

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
Also add `doc=` for all of the `declareProperty()` calls so the user docs have information

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**